File: app/src/main/java/com/letianpai/robot/desktop/ui/view/RobotTopView.java
package com.letianpai.robot.desktop.ui.view;

import android.content.Context;
import android.util.AttributeSet;

import com.letianpai.robot.desktop.R;
import com.letianpai.robot.desktop.ui.base.AbstractMainView;
import com.letianpai.robot.desktop.ui.base.TopView;

public class RobotTopView extends TopView {

    public RobotTopView(Context context) {
        super(context);
    }

    public RobotTopView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RobotTopView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public RobotTopView(AbstractMainView view, Context context) {
        super(view, context);
    }

    @Override
    protected void initView() {
        super.initView();
        setShutDirect(DOWN_TO_UP);
        inflate(mContext, R.layout.robot_top_view, this);
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/ui/view/RobotTopView.kt
package com.letianpai.robot.desktop.ui.view

import android.content.Context
import android.util.AttributeSet
import com.letianpai.robot.desktop.R
import com.letianpai.robot.desktop.ui.base.AbstractMainView
import com.letianpai.robot.desktop.ui.base.BaseView
import com.letianpai.robot.desktop.ui.base.TopView

class RobotTopView : TopView {
    constructor(context: Context?) : super(context)

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    constructor(view: AbstractMainView?, context: Context?) : super(view, context)

    override fun initView() {
        super.initView()
        setShutDirect(DOWN_TO_UP)
        inflate(mContext, R.layout.robot_top_view, this)
    }
}
